tokio/src/sync/watch.rs

```diff
@@ -437,6 +437,36 @@ impl<T> Sender<T> {
         Ok(())
     }
 
+    /// Modifies watched value, notifying all receivers.
+    ///
+    /// This can useful for modifying the watched value, without
+    /// having to allocate a new instance. Additionally, this
+    /// method permits sending values even when there are no receivers.
+    ///
+    /// # Examples
+    /// ```
+    /// use tokio::sync::watch;
+    ///
+    /// struct State {
+    ///     counter: usize,
+    /// }
+    /// let (state_tx, state_rx) = watch::channel(State { counter: 0 });
+    /// state_tx.send_modify(|state| state.counter += 1);
+    /// assert_eq!(state_rx.borrow().counter, 1);
+    /// ```
+    pub fn send_modify<F>(&self, func: F)
+    where
+        F: FnOnce(&mut T),
+    {
+        {
+            let mut lock = self.shared.value.write().unwrap();
+            func(&mut lock);
+            self.shared.state.increment_version();
+        }
```

Comment:
The `send_replace` method has a bunch of comments in this region. Please include them.

Also, it seems like we should be able to replace the implementation of `send_replace` with a call to `send_modify`.

Reply:
I've copied the comments from `send_replace`. In order to replace `send_replace` implementation with `send_modify` call to send_replace would have to take `mut value: T` in order to use `std::mem::swap`. `FnOnce(&mut T) -> T` doesn't make much sense to me. Am I missing something obvious or `send_replace` taking `mut value` would not be a breaking change.

Reply:
Changing it to be `mut value: T` is not actually a change of the signature, but if you feel better about it, you can add this at the top of the method:
```Rust
let mut value = value;
```

Reply:
I just wasn't sure whether it'd be a breaking change. Thanks.